hasher-matcher-actioner/hmalib/lambdas/actions/action_evaluator.py

```diff
@@ -78,15 +86,29 @@ def lambda_handler(event, context):
 
         logger.info("Evaluating against action_rules: %s", action_rules)
 
-        action_label_to_action_rules = get_actions_to_take(match_message, action_rules)
+        submitted_content = ContentObject.get_from_content_id(
+            config.dynamo_db_table,
+            match_message.content_key,
+        )
+
+        # Convert to dict becuase JSON can't serialize Sets for Action Message
+        # plus we think of these fields as key: value anyways
+        additional_fields: t.Dict[str, str] = dict(
+            field.split(":", 1) for field in submitted_content.additional_fields
```

Comment:
I'm worried about adding this dependency without changes to the API, nothing requires there be a `:` as part of this set nor that the prefix have a unique property. I viewed these fields as more of a set of tags/labels on a piece of content not a key-value pair, this is a disconnect between the UI and the API/underlying data model we should fix. 

Also, unfortunately as it is written I think this will break the action evaluator for content without any additional_fields

Reply:
I see. I thought the Submit API took a dict but I'm seeing now its a List. I'll change this to convert the set to a list as well.

I do think we should change the UI to match this data model

Reply:
Done: https://github.com/facebook/ThreatExchange/pull/745

Reply:
@BarrettOlson Updated to make a list of strings